### PR TITLE
cranelift: Remove EBB references from docs

### PR DESCRIPTION
**Pull request by elliottt**
Remove some references in the docs to extended basic blocks, now that we've completely migrated away.

### CODE REVIEW

File: cranelift/docs/ir.md [hunk 1]
```diff
@@ -311,23 +311,23 @@ Signaling NaNs
 
 ## Control flow
 
-Branches transfer control to a new EBB and provide values for the target EBB's
+Branches transfer control to a new BB and provide values for the target BB's
 arguments, if it has any. Conditional branches only take the branch if their
 condition is satisfied, otherwise execution continues at the following
-instruction in the EBB.
+instruction in the BB.
```
Comment (cfallin) [inline cranelift/docs/ir.md:317]: Here we can update w.r.t. the two-target conditional branches that we now use, I think?

**Review by cfallin — APPROVED** [review]
LGTM! I did a quick skim for other obviously outdated content and noted two-target conditionals below; also there's a stray `b1` around line 241 if you want to get that here. (We should make a more detailed pass on this and other docs but this PR isn't the time for that!)

File: cranelift/docs/ir.md (hunk 1 above)
Comment (elliottt) [reply cranelift/docs/ir.md:317]: Good catch!

File: cranelift/docs/ir.md [hunk 2]
```diff
@@ -311,23 +311,23 @@ Signaling NaNs
...
 
-JT = jump_table [EBB0, EBB1, ..., EBBn]
+JT = jump_table [BB0, BB1, ..., BBn]
     Declare a jump table in the [function preamble].
```
Comment (jameysharp) [inline cranelift/docs/ir.md:320]: We don't declare jump tables in the function preamble any more either, do we?

Comment (elliottt) [reply cranelift/docs/ir.md]: I just removed that after noticing the same thing by @cfallin's comment :D

**Pull request merged**